cli/lib/kontena/cli/stacks/stacks_helper.rb

```diff
@@ -69,19 +69,15 @@ def wait_for_deploy_to_finish(deployment, timeout = 600)
       raise 'deploy timed out'
     end
 
-    def wait_for_service_deploy(service_deploy, states)
+    def wait_for_service_deploy(service_deploy)
       service_deployed = false
       name = service_deploy['service_id'].split('/')[-1]
       spinner "Deploying service #{pastel.cyan(name)}" do |spin|
-        until service_deployed
-          r = client.get("services/#{service_deploy['service_id']}/deploys/#{service_deploy['id']}")
-          if states.include?(r['state'])
-            service_deployed = true
-            spin.fail if r['state'] == 'error'
-          else
-            sleep 1
-          end
+        until service_deploy['finished_at']
+          sleep 1
+          service_deploy = client.get("services/#{service_deploy['service_id']}/deploys/#{service_deploy['id']}")
```

Comment:
Now using the same logic as in the CLI [`ServicesHelper#wait_for_deploy_to_finish`](https://github.com/kontena/kontena/blob/6db8a3f89da6cca391d1c331d5ad4c3ae0e3ab79/cli/lib/kontena/cli/services/services_helper.rb#L290-L293). The two really need refactoring to share the same code, but that's over in #2195